src/App.tsx

```diff
@@ -1,61 +1,146 @@
 import './App.scss';
+import React, { FunctionComponent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './react-app-env';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUserById = (userId: number) => {
+  const foundedUser = usersFromServer.find(user => user.id === userId);
+
+  return foundedUser || null;
+};
+
+const todosWithUser: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.id),
+}));
+
+export const App: FunctionComponent = () => {
+  const [title, setTitle] = useState('');
+  const [authorId, setAuthorId] = useState(0);
+  const [todoList, setTodoList] = useState(todosWithUser);
+  const [titleError, setTitleError] = useState(false);
+  const [authorError, setAuthorError] = useState(false);
+  const id = Math.max(...todoList.map(todo => todo.id));
+  const getNewTodoId = () => (
+    id + 1
+  );
+  const clearForm = () => {
+    setTitle('');
+    setAuthorId(0);
+    setTitleError(false);
+    setAuthorError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const trimmedTitle = title.trim();
+
+    if (!trimmedTitle) {
+      setTitleError(() => true);
```

Comment:
if you don't use current state for defining new value, functional setState is redundant, just pass needed value as an argument to the setTitleError function